lib/fog/compute/ecloud.rb

```diff
@@ -356,7 +356,14 @@ def set_extra_headers_for(params)
             "x-tmrk-version" => @version,
             "Date"           => Time.now.utc.strftime("%a, %d %b %Y %H:%M:%S GMT"),
           }.merge(params[:headers] || {})
-          if params[:method] == "POST" || params[:method] == "PUT"
+          if params[:method] == "POST" || params[:method] == "PUT" || params[:method] == "DELETE"
+            if !params[:headers]['Content-Length']
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.